test/Unit/App/Middleware/DeprecationMiddlewareTest.php

```diff
@@ -185,6 +206,196 @@ public function process(
         $this->assertTrue($response->hasHeader('sunset'));
         $this->assertTrue($response->hasHeader('link'));
         $this->assertSame('2038-01-01', $response->getHeader('sunset')[0]);
-        $this->assertSame('test-link', $response->getHeader('link')[0]);
+        $this->assertSame($this->formatLink('test-link', [
+            'rel'  => 'test-rel',
+            'type' => 'test-type',
+        ]), $response->getHeader('link')[0]);
+    }
+
+    /**
+     * @throws ReflectionException
+     * @throws Exception
+     */
+    public function testDeprecationMethodUsesRequestMethod(): void
+    {
+        $handler = new class implements RequestHandlerInterface {
+            use HandlerTrait;
+
+            #[MethodDeprecation(
+                sunset: '2038-01-01',
+                link: 'get-test-link',
+                deprecationReason: 'get-test-deprecation-reason',
+                rel: 'get-rel',
+                type: 'get-type',
+            )]
+            public function get(ServerRequestInterface $request): ResponseInterface
```

Comment:
```suggestion
            public function get(): ResponseInterface
```
Unused parameter 'request'. The parameter's value is not used anywhere.